parcels/kernel.py

```diff
@@ -384,6 +384,8 @@ def remove_deleted(pset):
             # Apply recovery kernel
             for p in np.where(error_particles)[0]:
                 particles.set_index(p)
+                if particles.state == ErrorCode.StopExecution:
+                    return
```

Comment:
yes, simple and easy. That means that the manual stop of the execution still somehow needs to be attached to a particle, and it's limited kernel instruction set, right ? Still, it does match what #779 discusses and requests, and for that, this solution is nice.